core/common/test/LocalDateRangeTest.kt

```diff
@@ -0,0 +1,274 @@
+        )
+        assertEquals(
+            "2000-01-05 downTo 2000-01-01 step -1D",
+            LocalDateProgression.fromClosedRange(Jan_05_2000, Jan_01_2000, -1, DateTimeUnit.DAY).toString()
+        )
+    }
+
+    @Test
+    fun random() {
+        assertEquals(
+            Jan_01_2000,
+            (Jan_01_2000..Jan_01_2000).random()
+        )
+
+        assertEquals(
+            Jan_01_2000,
+            (Jan_01_2000 downTo Jan_01_2000).random()
+        )
+
+        assertFails {
+            (Jan_02_2000..Jan_01_2000).random()
+        }
+
+        assertNull((Jan_02_2000..Jan_01_2000).randomOrNull())
+
+        val seed = 123456
+        val expectedRand = Random(seed)
+        val actualRand = Random(seed)
+
+        repeat(20) {
+            assertEquals(
+                expectedRand.nextLong(0L..23L).let { Jan_01_2000.plus(it, DateTimeUnit.DAY) },
+                (Jan_01_2000..Jan_24_2000).random(actualRand)
+            )
+        }
+
+        repeat(20) {
+            assertEquals(
+                expectedRand.nextLong(0L..23L).let { Jan_24_2000.minus(it, DateTimeUnit.DAY) },
+                (Jan_24_2000 downTo Jan_01_2000).random(actualRand)
+            )
+        }
+
+        listOf(1L, 2L, 5L, 30L).forEach { step ->
+            repeat(20) {
+                val range = (0L..23L step step)
+                assertEquals(
+                    expectedRand.nextLong(0L..range.last / step).let { Jan_01_2000.plus(it * step, DateTimeUnit.DAY) },
+                    (Jan_01_2000..Jan_24_2000).step(step, DateTimeUnit.DAY).random(actualRand)
+                )
+            }
+
+            repeat(20) {
+                val range = (0L..23L step step)
+                assertEquals(
+                    expectedRand.nextLong(0..range.last / step).let { Jan_24_2000.minus(it * step, DateTimeUnit.DAY) },
+                    (Jan_24_2000 downTo Jan_01_2000).step(step, DateTimeUnit.DAY).random(actualRand)
+                )
+            }
+        }
+        repeat(20) {
+            assertTrue {
+                (Jan_01_2000..Jan_24_2000).step(5, DateTimeUnit.DAY).let { it.contains(it.random()) }
+            }
+        }
+    }
+
+    @Test
+    fun first() {
+        assertEquals((Jan_01_2000..Jan_01_2000).first(), Jan_01_2000)
+        assertEquals((Jan_01_2000 downTo Jan_01_2000).first(), Jan_01_2000)
+        assertEquals((Jan_01_2000..Jan_05_2000).first(), Jan_01_2000)
+        assertEquals((Jan_05_2000 downTo Jan_01_2000).first(), Jan_05_2000)
+        assertFails { (Jan_02_2000..Jan_01_2000).first() }
+        assertFails { (Jan_01_2000 downTo Jan_02_2000).first() }
+    }
+
+    @Test
+    fun last() {
+        assertEquals((Jan_01_2000..Jan_01_2000).last(), Jan_01_2000)
+        assertEquals((Jan_01_2000 downTo Jan_01_2000).last(), Jan_01_2000)
+        assertEquals((Jan_01_2000..Jan_05_2000).last(), Jan_05_2000)
+        assertEquals((Jan_05_2000 downTo Jan_01_2000).last(), Jan_01_2000)
```

Comment:
Also worth testing that `last()` returns not the specified right boundary but the actual last emitted element, like `(1..10).step(5).last() == 6`.